dozer/cogs/management.py

```diff
@@ -0,0 +1,209 @@
+"""General, basic commands that are common for Discord bots"""
+
+import asyncio
+import json
+import logging
+import math
+import os
+
+from datetime import timezone, datetime
+
+import humanize
+import discord
+from dateutil import parser
+from discord.ext.commands import has_permissions, CommandInvokeError
+
+from ..db import asyncpg
+from ._utils import *
+from .general import blurple
+from .. import db
+
+DOZER_LOGGER = logging.getLogger(__name__)
+
+timezone_file = "timezones.json"
+
+
+class Management(Cog):
+    """A cog housing Guild management/utility commands."""
+
+    def __init__(self, bot):
+        super().__init__(bot)
+        self.started_timers = False
+        self.timers = {}
+        if os.path.isfile(timezone_file):
+            DOZER_LOGGER.info("Loaded timezone configurations")
+            with open(timezone_file) as f:
+                self.timezones = json.load(f)
+        else:
+            DOZER_LOGGER.error("Unable to load timezone configurations")
+            self.timezones = {"abbreviations": []}
+
+    @Cog.listener('on_ready')
+    async def on_ready(self):
+        """Restore time based event schedulers"""
+        messages = await ScheduledMessages.get_by()
+        started = 0
+        if not self.started_timers:
+            for message in messages:
+                task = self.bot.loop.create_task(self.msg_timer(message))
+                self.timers[message.request_id] = task
+                started += 1
+            self.started_timers = True
+            DOZER_LOGGER.info(f"Started {started}/{len(messages)} scheduled messages")
+        else:
+            DOZER_LOGGER.info("Client Resumed: Timers still running")
+
+    async def msg_timer(self, db_entry):
+        """Holds the futures for sending a message"""
+        delay = db_entry.time - datetime.now(tz=timezone.utc)
+        if delay.total_seconds() > 0:
+            await asyncio.sleep(delay.total_seconds())
+        await self.send_scheduled_msg(db_entry)
+        await db_entry.delete(request_id=db_entry.request_id)
+
+    async def send_scheduled_msg(self, db_entry, channel_override=None):
+        """Formats and sends scheduled message"""
+        embed = discord.Embed(title=db_entry.header if db_entry.header else "Scheduled Message", description=db_entry.content)
+        guild = self.bot.get_guild(db_entry.guild_id)
+        channel = guild.get_channel(db_entry.channel_id if not channel_override else channel_override)
+        embed.colour = blurple
+        if db_entry.requester_id:
+            name = await guild.fetch_member(db_entry.requester_id)
+            embed.set_footer(text=f"Author: {name.display_name}")
+        await channel.send(embed=embed)
+
+    @group(invoke_without_command=True)
+    @has_permissions(manage_messages=True)
+    async def schedulesend(self, ctx):
```

Comment:
This has the `invoke_without_command` parameter set to True, however the bare group has an empty body.